src/project/mod.rs

```diff
@@ -768,6 +819,36 @@ pub fn ensure_toml_target_table<'a>(
         })
 }
 
+/// Retrieve a target table (`table_name`) inside a platform-specific table if everything
+/// goes as expected.
+pub fn get_toml_target_table<'a>(
+    doc: &'a mut Document,
+    platform: Platform,
+    table_name: &str,
+) -> miette::Result<&'a mut Table> {
+    let platform_table = doc["target"][platform.as_str()]
+        .as_table_mut()
+        .ok_or_else(|| {
+            miette::miette!(
+                "{}Could not find {} table in {}",
+                console::style(console::Emoji("❌ ", "X")).red(),
```

Comment:
The X is only really needed when you directly print the line to the console. When a `miette::Report` is printed the X and stuff are already added.